### DISCUSSION TITLE
Maintenance and Support for Angular versions

### DISCUSSION THREAD

**Discussion by MGREMY** [themesberg/flowbite-angular#61] [Ideas]
---
From issue n°#54 moved here
Author : @tomavic
---

### The question is How far can you support Angular versions? 

However, After checking discord and Figma I can see there are lots of work pending. This Answer to this will narrow the amount of work needed. 

In addition to the work pending such as adding components, accessibility and so on, there are developer-experience-related enhancements and refactors needed. 

From my side I will try to list them here, Please feel free to add yours too.

**Comment by MGREMY** [comment]
---
From https://github.com/themesberg/flowbite-angular/issues/54#issuecomment-2379200780
---

## The first issue: No Secondary Entry Point for Components/Directives

As stated from [ng-packager](https://github.com/ng-packagr/ng-packagr/blob/main/docs/secondary-entrypoints.md)

> Besides the primary entry point, a package can contain one or more secondary entry points (e.g. @angular/core/testing, @angular/cdk/a11y, …). These contain symbols that we don't want to group together with the symbols in the main entry point. The module id of a secondary entry directs the module loader to a sub-directory by the secondary's name.

For library developers, secondary entry points are configured by adding a ng-package.json file in subdirectories of the main package.json file's folder!

### Before
<img width="445" alt="Screenshot 2024-09-27 at 3 36 14 PM" src="https://github.com/user-attachments/assets/dc23ae96-0d76-4bf9-afd5-432337baf370">

### After
<img width="432" alt="Screenshot 2024-09-27 at 3 40 46 PM" src="https://github.com/user-attachments/assets/32d02be0-55e7-4e07-9753-a0ce1cdbaa25">

and the library consumer will use it like this

```ts

import { SanitizeHtmlPipe } from 'flowbite-angular/pipes'
```

and maybe we can make it better be adding each components independently inside the src folder without need to group them (components, services, driectives) and so on. 

```ts

import { Button } from 'flowbite-angular/button'
```

**Comment by MGREMY** [comment]
Hey @tomavic
Are you on the flowbite's discord, I might need some help to implement a clean ng-packagr setup
So if you are on discord, send me a message ! :)

**Reply by tomavic** [reply]
Sure I will do
